src/modal/index.js

```diff
@@ -22,60 +22,53 @@ class Modal extends Component {
     this.setState({
       open: false,
     });
-  }
+  };
 
   render() {
     const {
-      theme, className, body, title, footer, children, closeOnBackdropClick,
+      theme,
+      className,
+      body,
+      title,
+      footer,
+      children,
+      closeOnBackdropClick,
     } = this.props;
     const { open } = this.state;
-    const classes = cx(theme.modal, open === true ? theme['d-block'] : theme['d-none'], className);
-    const backdrop = cx(theme['modal-backdrop'], open === true ? theme['d-block'] : theme['d-none']);
-    const childelements = title || footer
-      ? (
-        <div className={theme.classes}>
-          {
-            title
-            ? (
-              <div className={theme['modal-title']}>
-                { title || '' }
-              </div>
-            )
-            : null
-          }
-          {children}
-          {
-            footer
-          ? (
-            <div className={theme['modal-footer']}>
-              { footer }
-            </div>
-          )
-          : null
-          }
-        </div>
-      ) : children;
+    const classes = cx(
+      theme.modal,
+      open === true ? theme['d-block'] : theme['d-none'],
+      className,
+    );
+    const backdrop = cx(
+      theme['modal-backdrop'],
+      open === true ? theme['d-block'] : theme['d-none'],
+    );
     /* eslint-disable jsx-a11y/no-static-element-interactions */
     /* eslint-disable jsx-a11y/click-events-have-key-events */
     return (
-      <div className={backdrop} onClick={closeOnBackdropClick ? this.closeModal : undefined}>
-        <div className={classes}>
-          {
-           childelements ||
-           <div>
-             <div className={theme['modal-title']}>
-               { title || '' }
-             </div>
-             <div className={theme['modal-body']}>
-               { body || '' }
-             </div>
-             <div className={theme['modal-footer']}>
-               { footer || '' }
-             </div>
-           </div>
-          }
-          <div className={theme.close} onClick={this.closeModal} />
-        </div>
+      <div
+        className={backdrop}
+        onClick={closeOnBackdropClick ? this.closeModal : undefined}
+      >
+        {open && (
+          <div id="modal" className={classes}>
```

Comment:
Readability looks to be little less. Can you split this into sub render functions ?

Reply:
Sure. I'll make changes and push.